src/hydra-eval-jobs/hydra-eval-jobs.cc

```diff
@@ -504,6 +504,16 @@ int main(int argc, char * * argv)
 
             job.erase("namedConstituents");
 
+            /* Register the derivation as a GC root.  !!! This
+                registers roots for jobs that we may have already
+                done. */
+            auto localStore = store.dynamic_pointer_cast<LocalFSStore>();
+            if (gcRootsDir != "" && localStore) {
+                Path root = gcRootsDir + "/" + std::string(baseNameOf((std::string) job["drvPath"]));
+                if (!pathExists(root))
+                    localStore->addPermRoot(localStore->parseStorePath((std::string) job["drvPath"]), root);
+            }
+
```

Comment:
Pretty much pasta'd from https://github.com/NixOS/hydra/blob/290e0653ad0121b80e53dfc0a5f9eed7540049ff/src/hydra-eval-jobs/hydra-eval-jobs.cc#L201-L209 but I'm anxious about creating a bigger diff here without any test framework on the C++ code.

Reply:
I wonder what it would take to implement a test framework...

Reply:
Yeah, good question. Here's a "My First Test" level of implementation: https://github.com/andir/hydra/commit/ba51556573e983c0baac84cf009b725769a1cac1

Reply:
Can we pull this out in a function?

Reply:
If the issue is the variables it closes over, we can do a C++ lambda too I am happy to help.